com.trekglobal.idempiere.rest.api/src/com/trekglobal/idempiere/rest/api/v1/resource/impl/ModelResourceImpl.java

```diff
@@ -430,59 +430,25 @@ public Response create(String tableName, String jsonText) {
 						.entity(new ErrorBuilder().status(Status.INTERNAL_SERVER_ERROR).title("Save error").append("Save error with exception: ").append(ex.getMessage()).build().toString())
 						.build();
 			}
-			
 			Map<String, JsonArray> detailMap = new LinkedHashMap<>();
 			Set<String> fields = jsonObject.keySet();
 			for(String field : fields) {
-				JsonElement fieldElement = jsonObject.get(field);
-				if (fieldElement != null && fieldElement.isJsonArray()) {
-					MTable childTable = MTable.get(Env.getCtx(), field);
-					if (childTable != null && childTable.getAD_Table_ID() > 0) {
-						IPOSerializer childSerializer = IPOSerializer.getPOSerializer(field, MTable.getClass(field));
-						JsonArray fieldArray = fieldElement.getAsJsonArray();
-						JsonArray savedArray = new JsonArray();
-						try {
-							fieldArray.forEach(e -> {
-								if (e.isJsonObject()) {
-									JsonObject childJsonObject = e.getAsJsonObject();
-									PO childPO = childSerializer.fromJson(childJsonObject, childTable);
-									childPO.set_TrxName(trx.getTrxName());
-									childPO.set_ValueOfColumn(tableName+"_ID", po.get_ID());
-									fireRestSaveEvent(childPO, PO_BEFORE_REST_SAVE, true);
-								if (! childPO.validForeignKeys()) {
-										String msg = CLogger.retrieveErrorString("Foreign key validation error");
-										throw new AdempiereException(msg);
-									}
-									childPO.saveEx();
-									fireRestSaveEvent(childPO, PO_AFTER_REST_SAVE, true);
-									childJsonObject = serializer.toJson(childPO);
-									savedArray.add(childJsonObject);
-								}
-							});
-							if (savedArray.size() > 0)
-								detailMap.put(field, savedArray);
-						} catch (Exception ex) {
-							trx.rollback();
-							log.log(Level.SEVERE, ex.getMessage(), ex);
-							return Response.status(Status.INTERNAL_SERVER_ERROR)
-									.entity(new ErrorBuilder().status(Status.INTERNAL_SERVER_ERROR).title("Save error").append("Save error with exception: ").append(ex.getMessage()).build().toString())
-									.build();
-						}
-					}
-				}
+				String strError = createChild(field, jsonObject, po, detailMap, trx);
+				if(strError != null)
+					return Response.status(Status.INTERNAL_SERVER_ERROR)
+							.entity(new ErrorBuilder().status(Status.INTERNAL_SERVER_ERROR).title("Can't perform document action").append("Encounter exception during execution of document action: ").append(strError).build().toString())
```

Comment:
Hi, the copy/paste oif this error message was wrong. The original error in this part of the code was: 
.entity(new ErrorBuilder().status(Status.INTERNAL_SERVER_ERROR).title("Save error").append("Save error with exception: ").append(ex.getMessage()).build().toString())

To throw a document action error here is missleading.

Reply:
Hi Diego, Thanks, I corrected it